### ISSUE TITLE
Irrelevant Quests showing

### ISSUE THREAD

**Issue by chuman72486** [IrcDirk/Carbonite-Classic#166]
I think this might be a duplicate issue, but I couldn't find anything that specifically lists this, so I figure I will post here.

In Carbonite, it shows quest markers, which I assume are meant for quests that should be available to start, however I keep finding cases where the "!" icon shows where there are no relevant quests.  For example, quests I've already completed or are not eligible for, quests from the other faction, or quests that require a certain reputation level or other requirements to appear.

I can understand not being able to filter out quests that have special requirements besides a prerequisite quest, but it's really misleading when I'm looking at my map littered with "!" icons and they're all Horde quests while I'm on my Alliance character.  I keep getting my hopes up that there's a bunch of quests I can do and I nearly run into an enemy town looking for it.
I'm also finding that it's showing quests I'm too low level to pick up.  My character is currently lv63 and I find quests marks where the only remaining quests are lv70 quests.

Examples posted below:
Quest Marker where only lv70 quests are "available" (I'm lv63)
Quest Marker where quests I can't get yet because I'm still on a previous quest are "available"
and Map of an area showing where I have a bunch of Horde quests showing "available" while I'm on an Alliance character

![image](https://user-images.githubusercontent.com/8305657/120930778-be130180-c6bc-11eb-98fe-9580a93bb629.png)

**Comment by IrcDirk** [comment]
Yeah, TBC API is based on Shadowlands API but without some functions. I need to find right one to check if quest is Ally or Horde. Or if not ill scrap the data from quests source.

**Comment by chuman72486** [comment]
First thing I've noticed is actually that TBC Classic appears to still be using 8.1.0 API, as all new 9.0.1 functions appear to be missing and many functions that were removed in 9.0+, but existed in 8.1 still work in TBC Classic.
For example, `C_QuestLog.GetTitleForQuestID()` (Added 9.0.1) does not work, but `C_QuestLog.GetQuestInfo()` (Added 8.0.1, Removed 9.0.1) does.
On that note, I could not find a single function anywhere in the API that suggests if a quest belongs to a specific faction, nor a way to determine if a specific player meets the requirements to start a quest.  This likely means all quests in the database will need to be tagged manually with their faction requirements and prerequisite quests.
